tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -5,17 +5,30 @@
 
 import contextlib
 import functools
+import logging
+import sys
+from collections import namedtuple
+from threading import Lock
 
 from .base import ReplayableTest
 from .utilities import create_random_name, is_text_payload, trim_kwargs_from_test_function
 from .recording_processors import RecordingProcessor
 from .exceptions import AzureNameError
 
+_logger = logging.getLogger("preparer")
+_logger.setLevel(logging.WARNING)
+handler = logging.StreamHandler(stream=sys.stdout)
+handler.setFormatter(logging.Formatter('%(asctime)s %(name)-12s %(levelname)-8s %(message)s'))
+_logger.addHandler(handler)
```

Comment:
I usually don't like setting actual logger configuration hard-coded, I feel it narrows too much usage, especially since you can do pytest --cli-log-level and stuff like to do it for us for free

Reply:
++ Agreed, per our OOB discussion this was my completely ignoring best practices to force preparer errors to be shown, but I should have been more proactive and adjusted the top-level-exception-surfacing to unroll inner exceptions and log messages so that ideally the key parts of this (letting the user know about the nature of a critical preparer failure) come for free and without mucking with logging config.